rustls/src/crypto/mod.rs

```diff
@@ -186,6 +206,66 @@ pub struct CryptoProvider {
     pub key_provider: &'static dyn KeyProvider,
 }
 
+impl CryptoProvider {
+    /// Sets this `CryptoProvider` as the default for this process.
+    ///
+    /// This can be called successfully at most once in any process execution.
+    ///
+    /// Call this early in your process to configure which provider is used for
+    /// the provider.  The configuration should happen before any use of
+    /// [`ClientConfig::builder()`] or [`ServerConfig::builder()`].
+    pub fn install_as_process_default(self) -> Result<(), Arc<Self>> {
+        PROCESS_DEFAULT_PROVIDER.set(Arc::new(self))
+    }
+
+    /// Returns the default `CryptoProvider` for this process.
+    ///
+    /// This will be `None` if no default has been set yet.
+    pub fn process_default() -> Option<&'static Arc<Self>> {
+        PROCESS_DEFAULT_PROVIDER.get()
+    }
+
+    /// An internal function that:
+    ///
+    /// - gets the process_default(), or
+    /// - installs one `from_crate_features()`, or else
+    /// - panics about the need to call [`CryptoProvider::install_as_process_default()`]
+    pub(crate) fn process_default_or_default_from_crate_features() -> &'static Arc<Self> {
+        match Self::process_default() {
```

Comment:
Given that the `None` branch is a little involved, maybe consider writing this as

```rust
if let Some(provider) = Self::process_default() {
    return provider;
}

..
```

?

Reply:
Yeah that's nicer!